### PR TITLE
Add Cat & Dog

### PR DESCRIPTION
**Pull request by kemie**
Added two new icons for cat & dog
Contributes to #498 

Lucid  2022-10-19 at 153056

### CODE REVIEW

**Review by ericfennis — COMMENTED** [review]
@kemie Nice work!

According to our design guidelines we like to have 2px between elements.
See [Lucide Design principes](https://lucide.dev/docs/icon-design-guide#8-distinct-elements-must-have-2-pixels-of-spacing-between-each-other)
image

Is it possible to fix it?

**Comment by kemie** [comment]
I've fixed both icons to better comply with the guidelines. have checked them with 3px width and they seem to work well. 
Lucid  2022-10-25 at 092442@2x


I've updated my branch, is it necessary to do another pull request? (github newbie here)

**Comment by karsa-mistmere** [comment]
Really nice work, @kemie!

I think the only issue left is the lack of a gap between the dog's ears and its outline:
image

I don't necessarily think we need to have a 2px gap there, as the icon still remains perfectly legible with a small gap, but can you please make a version that has a somewhat larger gap there? (No need to update the PR with it just yet).

And no, it's fine, you do not need to create a new PR for every change, it updates automatically if you change the source branch. ^_^

**Comment by kemie** [comment]
hehe, as soon as I posted the image I realized that and fixed. How's this?
Lucid  2022-10-25 at 093137@2x

File: tags.json [hunk 1]
```diff
@@ -1318,6 +1322,10 @@
   "divide-square": [
     "calculate"
   ],
+  "dog": [
+    "animal",
+    "pet"
```
Comment (karsa-mistmere) [inline tags.json:1327]:
```suggestion
    "pet",
    "puppy",
    "hound",
    "canine"
```

File: tags.json [hunk 2]
```diff
@@ -763,6 +763,10 @@
     "chromecast",
     "airplay"
   ],
+  "cat": [
+    "animal",
+    "pet"
```
Comment (karsa-mistmere) [inline tags.json:768]:
```suggestion
    "pet",
    "kitten",
    "feline"
```

File: icons/cat.svg [hunk 3]
```diff
@@ -0,0 +1 @@
+<svg width="24" height="24" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M14 5.256A8.148 8.148 0 0 0 12 5a9.04 9.04 0 0 0-2 .227M20.098 10c.572 1.068.902 2.24.902 3.444C21 17.89 16.97 21 12 21s-9-3-9-7.556c0-1.251.288-2.41.792-3.444m-.042 0S2.11 3.58 3.5 3C4.89 2.42 8 3 9.781 5m10.391 5.002s1.64-6.42.25-7c-1.39-.58-4.5 0-6.282 2M8 14v.5m8-.5v.5" stroke="#000" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/><path d="M11.25 16.25h1.5L12 17l-.75-.75Z" stroke="#000" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/></svg>
```
Comment (karsa-mistmere) [inline icons/cat.svg:1]: The design itself is perfect as it is, I think, but the formatting is not quite right, be sure to follow the same [global attributes](https://lucide.dev/docs/icon-design-guide#global-attributes) and the same format as other icons, you can also optimize it using [SVGOMG](https://jakearchibald.github.io/svgomg/). For example:
```suggestion
<svg
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="M14 5.256A8.148 8.148 0 0 0 12 5a9.04 9.04 0 0 0-2 .227M20.098 10c.572 1.068.902 2.24.902 3.444C21 17.89 16.97 21 12 21s-9-3-9-7.556c0-1.251.288-2.41.792-3.444m-.042 0S2.11 3.58 3.5 3C4.89 2.42 8 3 9.781 5m10.391 5.002s1.64-6.42.25-7c-1.39-.58-4.5 0-6.282 2" />
  <path d="M8 14v.5m8-.5v.5" />
  <path d="M11.25 16.25h1.5" />
  <path d="M12 17l-.75-.75Z" />
</svg>
```

**Comment by karsa-mistmere** [comment]
Nice work, I think it now looks much better on 100%. I've made a few suggestions, if those are resolved, I think we can merge this.

**Comment by kemie** [comment]
think it's done!

**Comment by karsa-mistmere** [comment]
The SVG file formats still seem off to me, the attributes are in a different format than in our guideline and there are still some inline stroke attributes.

**Comment by kemie** [comment]
ok. I *think* it's right now

**Comment by karsa-mistmere** [comment]
We're almost there, we're just missing a few spaces and line breaks, see: https://github.com/lucide-icons/lucide/blob/main/icons/activity.svg?short_path=2eccd9c

**Comment by kemie** [comment]
now? I really appreciate your patience in walking me through all of this. For future reference are there any specific SVOMG settings I should be using to get the right output?

**Comment by karsa-mistmere** [comment]
I'm not sure if there's a better way, maybe @ericfennis has a better pipeline than I do, but I think you're best off using the global attributes as a template, inserting your paths from your working copy and running the command line svgo tool included with Lucide (`pnpm run optimize`).

File: icons/cat.svg [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  width="24"
+  height="24"
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+<path d="M14 5.256A8.148 8.148 0 0 0 12 5a9.04 9.04 0 0 0-2 .227M20.098 10c.572 1.068.902 2.24.902 3.444C21 17.89 16.97 21 12 21s-9-3-9-7.556c0-1.251.288-2.41.792-3.444m-.042 0S2.11 3.58 3.5 3C4.89 2.42 8 3 9.781 5m10.39 5.002s1.64-6.42.25-7c-1.39-.58-4.5 0-6.28 2M8 14v.5m8-.5v.5m-3.25 1.75h-1.5L12 17l.75-.75Z"/>
```
Comment (karsa-mistmere) [inline icons/cat.svg:12]: One more thing, you should break up compound paths into simple paths, as some apps render compound paths in pretty funky ways. I'm guessing this was an unintended side effect of SVGOMG having the merge paths setting on.

(I've tried this with Figma, possibly our biggest offender and it looks completely fine, but we can never be too careful :) ).

File: icons/cat.svg [hunk 5]
```diff
@@ -9,5 +9,10 @@
   stroke-linecap="round"
   stroke-linejoin="round"
 >
-<path d="M14 5.256A8.148 8.148 0 0 0 12 5a9.04 9.04 0 0 0-2 .227M20.098 10c.572 1.068.902 2.24.902 3.444C21 17.89 16.97 21 12 21s-9-3-9-7.556c0-1.251.288-2.41.792-3.444m-.042 0S2.11 3.58 3.5 3C4.89 2.42 8 3 9.781 5m10.39 5.002s1.64-6.42.25-7c-1.39-.58-4.5 0-6.28 2M8 14v.5m8-.5v.5m-3.25 1.75h-1.5L12 17l.75-.75Z"/>
+<path d="M14 5.256A8.148 8.148 0 0 0 12 5a9.04 9.04 0 0 0-2 .227M20.098 10c.572 1.068.902 2.24.902 3.444C21 17.89 16.97 21 12 21s-9-3-9-7.556c0-1.251.288-2.41.792-3.444"/>
+<path d="M3.75 10S2.11 3.58 3.5 3C4.89 2.42 8 3 9.781 5"/>
+<path d="M20.172 10.002s1.64-6.42.25-7c-1.39-.58-4.5 0-6.282 2"/>
+<path d="M8 14v.5"/>
+<path d="M16 14v.5"/>
+<path d="M11.25 16.25h1.5L12 17l-.75-.75Z"/>
```
Comment (karsa-mistmere) [inline icons/cat.svg:17]:
```suggestion
  <path d="M14 5.256A8.148 8.148 0 0 0 12 5a9.04 9.04 0 0 0-2 .227M20.098 10c.572 1.068.902 2.24.902 3.444C21 17.89 16.97 21 12 21s-9-3-9-7.556c0-1.251.288-2.41.792-3.444"/>
  <path d="M3.75 10S2.11 3.58 3.5 3C4.89 2.42 8 3 9.781 5"/>
  <path d="M20.172 10.002s1.64-6.42.25-7c-1.39-.58-4.5 0-6.282 2"/>
  <path d="M8 14v.5"/>
  <path d="M16 14v.5"/>
  <path d="M11.25 16.25h1.5L12 17l-.75-.75Z"/>
```

File: icons/dog.svg [hunk 6]
```diff
@@ -9,5 +9,10 @@
   stroke-linecap="round"
   stroke-linejoin="round"
 >
-<path d="M10 5.172C10 3.782 8.423 2.679 6.5 3c-2.823.47-4.113 6.006-4 7 .08.703 1.725 1.722 3.656 1 1.261-.472 1.96-1.45 2.344-2.5m5.767-3.328c0-1.39 1.577-2.493 3.5-2.172 2.823.47 4.113 6.006 4 7-.08.703-1.725 1.722-3.656 1-1.261-.472-1.855-1.45-2.239-2.5M8 14v.5m8-.5v.5M4.42 11.247A13.152 13.152 0 0 0 4 14.556C4 18.728 7.582 21 12 21s8-2.272 8-6.444c0-1.061-.162-2.2-.493-3.309m-9.243-6.082A8.801 8.801 0 0 1 12 5c.78 0 1.5.108 2.161.306M12.75 16.25h-1.5L12 17l.75-.75Z"/>
+<path d="M10 5.172C10 3.782 8.423 2.679 6.5 3c-2.823.47-4.113 6.006-4 7 .08.703 1.725 1.722 3.656 1 1.261-.472 1.96-1.45 2.344-2.5"/>
+<path d="M14.267 5.172c0-1.39 1.577-2.493 3.5-2.172 2.823.47 4.113 6.006 4 7-.08.703-1.725 1.722-3.656 1-1.261-.472-1.855-1.45-2.239-2.5"/>
+<path d="M8 14v.5"/>
+<path d="M16 14v.5"/>
+<path d="M11.25 16.25h1.5L12 17l-.75-.75Z"/>
+<path d="M4.42 11.247A13.152 13.152 0 0 0 4 14.556C4 18.728 7.582 21 12 21s8-2.272 8-6.444c0-1.061-.162-2.2-.493-3.309m-9.243-6.082A8.801 8.801 0 0 1 12 5c.78 0 1.5.108 2.161.306"/>
```
Comment (karsa-mistmere) [inline icons/dog.svg:17]:
```suggestion
  <path d="M10 5.172C10 3.782 8.423 2.679 6.5 3c-2.823.47-4.113 6.006-4 7 .08.703 1.725 1.722 3.656 1 1.261-.472 1.96-1.45 2.344-2.5"/>
  <path d="M14.267 5.172c0-1.39 1.577-2.493 3.5-2.172 2.823.47 4.113 6.006 4 7-.08.703-1.725 1.722-3.656 1-1.261-.472-1.855-1.45-2.239-2.5"/>
  <path d="M8 14v.5"/>
  <path d="M16 14v.5"/>
  <path d="M11.25 16.25h1.5L12 17l-.75-.75Z"/>
  <path d="M4.42 11.247A13.152 13.152 0 0 0 4 14.556C4 18.728 7.582 21 12 21s8-2.272 8-6.444c0-1.061-.162-2.2-.493-3.309m-9.243-6.082A8.801 8.801 0 0 1 12 5c.78 0 1.5.108 2.161.306"/>
```

**Review by karsa-mistmere — APPROVED** [review]

**Review by ericfennis — APPROVED** [review]
Awesome work @kemie !! Lovely icons 🐱 🐶

**Pull request merged**